File: src/main/java/dev/emortal/minestom/marathon/animator/NoAnimator.java
package dev.emortal.minestom.marathon.animator;

import net.minestom.server.coordinate.Point;
import net.minestom.server.instance.Instance;
import net.minestom.server.instance.block.Block;
import org.jetbrains.annotations.NotNull;

public final class NoAnimator implements BlockAnimator {

    @Override
    public void setBlockAnimated(@NotNull Instance instance, @NotNull Point point, @NotNull Block block, @NotNull Point lastPoint) {
        instance.setBlock(point, block);
    }

    @Override
    public void reset() {
        // do nothing
    }
}

File: src/main/java/dev/emortal/minestom/marathon/animator/NoAnimator.kt
package dev.emortal.minestom.marathon.animator

import net.minestom.server.coordinate.Point
import net.minestom.server.instance.Instance
import net.minestom.server.instance.block.Block

class NoAnimator : BlockAnimator {
    override fun setBlockAnimated(instance: Instance, point: Point, block: Block, lastPoint: Point) {
        instance.setBlock(point, block)
    }

    override fun reset() {
        // do nothing
    }
}
